theseus/optimizer/linear/tests/test_lu_cuda_sparse_solver.py

```diff
@@ -6,9 +6,7 @@
 import pytest  # noqa: F401
```

Comment:
I tried to run this in my computer, before I installed the CUDA extensions, and got an error as shown below. Do we want to make the tests in this file conditional on the extension being installed? We could use [pytest.importskip](https://docs.pytest.org/en/6.2.x/reference.html#pytest-importorskip) for this. 

![image](https://user-images.githubusercontent.com/4759586/148129174-adc69e32-8b46-4628-9212-6af2ab5d4820.png)

Reply:
well that's precisely what the ci currently does, they are only tested when the extension is installed.
However I would not make them run only if the extension is *available*, but rather on where the user tried to install with Cuda. In such a case they should fail and notify the user that something didn't work as expected and the extension she tried to compile is actually not available.

Reply:
So, that's right that running tests should not fail because theseus was correctly installed with no cuda support!
Not sure how that can be achieved, I was just saying that the test should just not skip because if was unable to load the extension. Ideally there should be a command running tests that will skip cuda tests if the extension is not installed, or two separate commands for the user to check both scenarios.
I think we have to minimize the confusion for the user, considering the two scenarios: 1. the user willingly disabled cuda extensions 2. the user wants cuda extensions, but they may not have installed properly, or be nonfunctional, and tests should check that. Suggestions here too @mhmukadam ?

Reply:
As discussed offline, let's make this test optional on depending on the extension being installed, and consider other mechanisms (e.g., a separate command) for testing that the installation was successful. Additionally, `pytest.importskip` has a `reason=` kwarg that we can use to print a warning message to alert those users that did in fact want to run the CUDA test that their install might be incorrect.

Reply:
in new revision I'm adding the "cuda" mark to cuda tests, and a note on the readme on how to skip those tests

Reply:
This is great!